_blogposts/compiler/a-story-of-lazy-encoding.mdx

```diff
@@ -0,0 +1,122 @@
+---
+author: hongbo
+date: "2020-05-15 18:00" 
+previewImg:
+category: compiler
+title: New Lazy Encoding in BuckleScript
+description: |
+  Highlights of our newest changes to the internal representation of exceptions
+  and how it will provide better stacktraces to our users.
+canonical: https://bucklescript.github.io/blog/2020/05/06/exception-encoding
+---
+
+---
+title: Enhanced lazy encoding in BuckleScript
+---
+
+
+
+Recently we made some significant improvements with our new encoding for lazy values, and we find it so exciting that we want to highlight the changes. The new encoding generates very idiomatic JS output like hand-written code.
+
+For people who are not familiar with lazy evaluation, it is documented here: https://ocaml.org/releases/4.10/htmlman/expr.html#sss:expr-lazy.
+
+# What's the difference?
+
+Take this code snippet, for example:
+
+```reasonml
```

Comment:
we usually use `reason` or `re` for reason snippets... since we are using `reasonml` quite a lot in existing sources, I will add the `reasonml` keyword in the whitelist as well. Currently on it, we'll need to rebase on master in a minute

Reply:
this is now fixed on master, as soon as you rebase it will build on our PR preview. Ignore my comment above and feel free to use `reasonml` from there on